src/modules/battery_status/battery_status.cpp

```diff
@@ -138,7 +139,19 @@ class BatteryStatus : public ModuleBase<BatteryStatus>, public ModuleParams, pub
 	uORB::Subscription	_vcontrol_mode_sub{ORB_ID(vehicle_control_mode)};		/**< vehicle control mode subscription */
 
 	orb_advert_t	_battery_pub[BOARD_NUMBER_BRICKS] {};			/**< battery status */
-	Battery		_battery[BOARD_NUMBER_BRICKS];			/**< Helper lib to publish battery_status topic. */
+
+	AnalogBattery1 _battery0;
+
+#if BOARD_NUMBER_BRICKS > 1
+	AnalogBattery2 _battery1;
```

Comment:
We should be able to split this into different modules by type and not have these ifdefs.

Reply:
I don't understand, these are all analog batteries in this module. The only other type of battery I have worked on is with an I2C power monitor in #12673 , and it is in a completely separate module. The reason these `#if`s are needed here is because, if a board only supports 1 power brick, but I declare both `AnalogBattery`s, then the compiler will complain about the unused variable.